engine/launcher/src/main/scala/org/enso/launcher/components/Manifest.scala

```diff
@@ -0,0 +1,125 @@
+package org.enso.launcher.components
+
+import java.io.{FileReader, IOException}
+import java.nio.file.Path
+
+import io.circe.{yaml, Decoder, DecodingFailure}
+import nl.gn0s1s.bump.SemVer
+import org.enso.launcher.Logger
+
+/**
+  * Version information identifying the runtime that can be used with an engine
+  * release.
+  *
+  * @param graal version of the GraalVM
+  * @param java Java version of the GraalVM flavour that should be used
+  */
+case class RuntimeVersion(graal: SemVer, java: String)
+
+/**
+  * Contains release metadata read from the manifest file that is attached to
+  * each release.
+  *
+  * @param minimumLauncherVersion The minimum required version of the launcher
+  *                               that can be used to run this engine release.
+  *                               Earlier launcher versions should not be able
+  *                               to download this release, but print a message
+  *                               that the launcher needs an upgrade.
```

Comment:
It should also inform users how to upgrade. In future it should probably ask them if they want it to do it for them.